app/Http/Controllers/CategoryController.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Http\Controllers;
+use App\Http\Requests\StoreCategoryRequest;
+use App\Http\Requests\UpdateCategoryRequest;
+use App\Services\CategoryService;
+use Illuminate\Http\Request;
+
+class CategoryController extends Controller
+{
+    protected $categoryService;
+
+    // Constructor to inject CategoryService
+    public function __construct(CategoryService $categoryService)
+    {
+        $this->categoryService = $categoryService;
+    }
+
+    // Get all categories
+    public function index()
+    {
+        $categories = $this->categoryService->getAllCategories();
+        return response()->json($categories);
+    }
+
+    // Get a category by ID
+    public function show($id)
+    {
+        $category = $this->categoryService->findCategoryById($id);
+        return response()->json($category);
+    }
+
+    // Create a new category
+
+    public function store(StoreCategoryRequest $request)
```

Comment:
Please use a DTO to send data instead of sending an array. Using a DTO makes it clear what data is being sent and enforces the types of the data, leading to better type safety, readability, and maintainability of the code.